### PR TITLE
refactor(electric): Refactor event processing in Replication.Postgres.MigrationConsumer

### PR DESCRIPTION
**Pull request by alco**
In trying to understand the logic in the code, I found it helpful to refactor it to use a more functional programming style. There's no change in behaviour.

I recommend reviewing this PR one commit at a time.

### CODE REVIEW

File: components/electric/lib/electric/replication/postgres/migration_consumer.ex [hunk 1]
```diff
@@ -134,75 +127,39 @@ defmodule Electric.Replication.Postgres.MigrationConsumer do
...
 
     table = %{table | primary_keys: pks}
 
-    state =
-      register_relation(table, columns, state)
-      |> refresh_subscription()
+    register_relation(table, columns, state)
+    |> refresh_subscription()
 
     # update the subscription to add any new
```
Comment (magnetised) [inline components/electric/lib/electric/replication/postgres/migration_consumer.ex:160]: nitpick: this comment is kind of orphaned -- should probably be moved above the `refresh_subscription` call

File: components/electric/lib/electric/replication/postgres/migration_consumer.ex [hunk 2]
```diff
@@ -4,7 +4,6 @@ defmodule Electric.Replication.Postgres.MigrationConsumer do
   """
   use GenStage
 
-  alias Ecto.Adapter.Transaction
```
Comment (magnetised) [inline components/electric/lib/electric/replication/postgres/migration_consumer.ex:7]: bloody lsp drops these things in. good catch.

**Review by magnetised — APPROVED** [review]

**Comment by magnetised** [comment]
@alco agreed - my version was hard to follow. would disagree with the assertion that your version is more "functional" but it's definitely more readable! thanks

**Comment by alco** [comment]
It's more functional in the sense that state updates are more obvious in the new version. Whereas before you had to track both `migrations` and `state` arguments down every clause of the `process_events()` function to see if any of them is updated there, in the new version there's less confusion about that.
